src/main/java/org/jenkinsci/plugins/configfiles/maven/job/MvnGlobalSettingsProvider.java

```diff
@@ -131,13 +134,20 @@ public String getDisplayName() {
             return "provided global settings.xml";
         }
 
-        public ListBoxModel doFillSettingsConfigIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project) {
-            project.checkPermission(Item.CONFIGURE);
-            
+        public ListBoxModel doFillSettingsConfigIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project, String currentValue) {
+            Permission permToCheck = project == null ? Jenkins.ADMINISTER : Item.EXTENDED_READ;
+            AccessControlled contextToCheck = project == null ? Jenkins.get() : project;
+
             ListBoxModel items = new ListBoxModel();
             items.add("please select", "");
+
+            if (!contextToCheck.hasPermission(permToCheck)) {
+                items.add("current", currentValue); // we just add what they send
```

Comment:
You are not marking this as selected, meaning the first value in the list will be used; the `please select`.

Reply:
Yes, good one. I changed that. It's for a user without permission to call this method, but it's better to do it good even so.